network-api/networkapi/wagtailpages/pagemodels/research_hub/detail_page.py

```diff
@@ -163,38 +172,52 @@ class ResearchDetailLink(wagtail_models.TranslatableMixin, wagtail_models.Ordera
     panels = [
         edit_handlers.HelpPanel(
             content=(
-                "Please provide an external link to the original source or upload a document and select it here. "
-                'If you wish to provide both, please create two separate "research links"'
+                "Please provide a link to the original resource. "
+                "You can link to an internal page, an external URL or upload a document. "
+                'If you wish to provide multiple, please create two separate "research links"'
             )
         ),
         edit_handlers.FieldPanel("label"),
         edit_handlers.FieldPanel("url"),
-        docs_handlers.FieldPanel("document"),
+        edit_handlers.FieldPanel("page"),
+        edit_handlers.FieldPanel("document"),
     ]
 
     class Meta(wagtail_models.TranslatableMixin.Meta, wagtail_models.Orderable.Meta):
         ordering = ["sort_order"]
 
-    def __str__(self):
+    def __str__(self) -> str:
         return self.label
 
-    def clean(self):
+    def clean(self) -> None:
         super().clean()
-        if self.url and self.document:
-            error_message = "Please provide either a URL or a document, not both."
+
+        # Ensure that only one of the three fields is set
+        if sum([bool(self.url), bool(self.page), bool(self.document)]) > 1:
+            error_message = "Please provide either a URL, a page or a document, not multiple."
             raise exceptions.ValidationError(
-                {"url": error_message, "document": error_message},
+                {"url": error_message, "page": error_message, "document": error_message},
                 code="invalid",
             )
-        elif not self.url and not self.document:
-            error_message = "Please provide a URL or a document."
+        # Ensure that at least one of the three fields is set
+        elif not any([self.url, self.page, self.document]):
+            error_message = "Please provide a URL, a page or a document."
             raise exceptions.ValidationError(
-                {"url": error_message, "document": error_message},
+                {"url": error_message, "page": error_message, "document": error_message},
                 code="required",
             )
 
-    def get_url(self):
+    def get_url(self) -> str:
         if self.url:
             return self.url
+        elif self.page:
+            if not self.page.live:
+                logger.warning(
+                    f"Detail link to unpublished page defined: { self } -> { self.page }. "
+                    "This link will not be shown in the frontend."
+                )
+                return ""
+            return self.page.get_url()
         elif self.document:
             return self.document.url
+        raise ValueError("No URL defined for this detail link. This should not happen.")
```

Comment:
@kevinhowbrook Mypy had failed before because this branch in the logic was not covered. I have added a test for this now too.